File: CohesiveRP.Core.WebApi/ResponseDtos/BackgroundQueries/BackgroundQueriesResponseDto.cs
using System.Net;
using System.Text.Json.Serialization;
using CohesiveRP.Common.WebApi;
using CohesiveRP.Core.WebApi.ResponseDtos.BackgroundQueries.BusinessObjects;

namespace CohesiveRP.Core.WebApi.ResponseDtos.BackgroundQueries
{
    public class BackgroundQueriesResponseDto : IWebApiResponseDto
    {
        [JsonConverter(typeof(JsonNumberEnumConverter<HttpStatusCode>))]
        [JsonPropertyName("code")]
        public HttpStatusCode HttpResultCode { get; set; }

        [JsonPropertyName("chatId")]
        public string ChatId { get; set; }

        [JsonPropertyName("queries")]
        public BackgroundQueryModel[] Queries { get; set; }
    }
}

File: CohesiveRP.UI.WebFrontend/src/ResponsesDto/chat/BackgroundQueriesResponseDto.ts
import type { ServerApiResponseDto } from "../ServerApiResponseDto";
import type { BackgroundQuery } from "./BusinessObjects/BackgroundQuery";

interface BackgroundQueriesResponseDto extends ServerApiResponseDto {
    chatId : string,
    queries: BackgroundQuery[]
}

export type {
    BackgroundQueriesResponseDto
};